### PR TITLE
checking for non existent keys

### PR DESCRIPTION
**Pull request by mdfleury-wbd**
Testing for non-existent keys so plugin doesn't throw errors

## How Has This Been Tested?
Tested in nonprod

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist

- [x] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/engine/plugins/node_dependencies/parse.py [hunk 1]
```diff
@@ -111,15 +111,15 @@ def _find_versions_v2(lockfile: dict, nodes: list, resolver: LineNumberResolver)
...
         pkg = lockfile["packages"][node]
         path = f"/{node}".replace("/node_modules/", ">")[1:]
-        source = resolver.find(pkg["integrity"]) or resolver.find(path)
+        source = resolver.find(pkg.get("integrity")) or resolver.find(path)
         versions.append(
             {
                 "path": path,
                 "version": pkg["version"],
-                "integrity": pkg["integrity"],
+                "integrity": pkg.get("integrity", ""),
```
Comment (g-marconet) [inline backend/engine/plugins/node_dependencies/parse.py:119]: Is empty string the best value for empty here? How would the rest of Artemis handle empty string as opposed to something like "None provided"?

(Not suggesting we change it here, just trying to understand. This also applies to `filename` below)

File: backend/engine/plugins/node_dependencies/parse.py [hunk 2]
```diff
@@ -111,15 +111,15 @@ def _find_versions_v2(lockfile: dict, nodes: list, resolver: LineNumberResolver)
...
                 "dev": pkg.get("dev", False),
-                "filename": source["filename"],
-                "line": source["line"],
+                "filename": source.get("filename", ""),
+                "line": source.get("line", ""),
```
Comment (g-marconet) [inline backend/engine/plugins/node_dependencies/parse.py:122]: What type is `line` when it does exist in `source`? If it's a number, does it make sense to sometimes have it be a string?

Comment (mdfleury-wbd) [reply backend/engine/plugins/node_dependencies/parse.py:122]: Yeah, it makes more sense as a string. I'll fix that.

File: backend/engine/plugins/node_dependencies/parse.py (hunk 1 above)
Comment (mdfleury-wbd) [reply backend/engine/plugins/node_dependencies/parse.py:119]: If you go up in this file, we never do anything with the integrity field, so it shouldn't change anything.

**Pull request merged**